### PR TITLE
Add base support for Synthetics

### PR DESCRIPTION
**Pull request by nic-lan**
#### The Goal

This PR aims to:
- add support for creating new API type synthetics
- add support for updating API type synthetics
- add support for get all the synthetics

#### The Why

I noticed that at the moment the gem is not providing any kind of support for the synthetics.

Based on that i follow the pattern defined on the documentation for the [datadog api](https://docs.datadoghq.com/api/?lang=ruby#synthetics) to provide the mimum support to automate the synthetics flow.
More synthetics operations can be added once this PR is approved.

#### Rubocop

While adding support for getting all the synthetics I found out that the current naming convention of gem about naming the reader is to prefifx with `get_` BUT this fact conflicts with the curent rubocop settings.
Indeed rubocop complained about:

```console
Do not prefix reader method names with get_.
```

I fixed that by escaping the synthetics file from the complaining rubocop and keeping the current naming convention of prefixing the reader with `get_`.

I did so because I don't think that the scope of this PR is to edit the rubocop settings and i suggest to either change them to deal with the current naming convention either to change the naming convention to deal with current rubocop settings.

Comments and suggestions are welcome.

### CODE REVIEW

**Comment by nic-lan** [comment]
still not merged ?

i can see 2 ❤️ .. is all that love not enough to merge the PR ?

**Comment by dabcoder** [comment]
Hi @nic-lan, apologies for the delay in getting back to you. We'll review this as soon as possible, bear with us but we will get back to you. Thanks for working on this.

File: lib/dogapi/v1/synthetic.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'dogapi'
+
+module Dogapi
+  class V1 # for namespacing
+    # SyntheticService is the class responsible for dealing with the synthetics
+    class SyntheticService < Dogapi::APIService
+
+      API_VERSION = 'v1'
+
+      def synthetic(type, config, options = {})
```
Comment (ericmustin) [inline lib/dogapi/v1/synthetic.rb]: small nit, prefer to name this `create_synthetic` to keep in line with current library naming convention. You can refer to `dashboard` for structure, here: `https://github.com/DataDog/dogapi-rb/blob/0032f0b63ce8f356e5e0cea6a0d33fafc1c63bcf/lib/dogapi/v1/dashboard.rb#L31` .  

Additionally generally we lay out the required and optional args here in code comments above, but this endpoint's args are pretty large and verbose, so perhaps maybe just a comment pointing to the doc links for required/optional args. `https://docs.datadoghq.com/api/?lang=bash#create-a-test`

 thoughts? deferring to @dabcoder here

File: lib/dogapi/facade.rb [hunk 2]
```diff
@@ -44,6 +44,7 @@ def initialize(api_key, application_key=nil, host=nil, device=nil, silent=true,
       @embed_svc = Dogapi::V1::EmbedService.new(@api_key, @application_key, silent, timeout, @datadog_host)
       @screenboard_svc = Dogapi::V1::ScreenboardService.new(@api_key, @application_key, silent, timeout, @datadog_host)
       @monitor_svc = Dogapi::V1::MonitorService.new(@api_key, @application_key, silent, timeout, @datadog_host)
+      @synthetic_svc = Dogapi::V1::SyntheticService.new(@api_key, @application_key, silent, timeout, @datadog_host)
```
Comment (ericmustin) [inline lib/dogapi/facade.rb]: remove `app/integration/synthetic.rb` as this  looks like it got pulled in from work unrelated to this repository?

File: lib/dogapi/v1/synthetic.rb (hunk 1 above)
Comment (dabcoder) [reply lib/dogapi/v1/synthetic.rb]: We called it `create_test` in `datadogpy`: https://github.com/DataDog/datadogpy/blob/c880facaf8c79927ab2933b713d7dfa64d8ac86b/datadog/api/synthetics.py#L135 to be more specific about what the method actually does.

**Comment by dabcoder** [comment]
Closing this as it will be superseded by #210. Thanks for working on this @nic-lan, we've kept your original commits in this aforementioned PR.

**Comment by nic-lan** [comment]
Ehi @dabcoder,

i was out for some time.
I am happy you found the code useful.
I will delete this branch so that you can go on with #210

**Pull request closed without merging**